traffic_monitor/cache/astats.go

```diff
@@ -116,8 +119,22 @@ func astatsParse(cacheName string, rdr io.Reader, pollCTX interface{}) (Statisti
 		astats.Ats["system.proc.loadavg"] = astats.System.ProcLoadavg
 		astats.Ats["system.proc.net.dev"] = astats.System.ProcNetDev
 
+		via := ctx.HTTPHeader.Get(rfc.Via)
+		if via != "" {
+			viaRegexSubmatch := regexp.MustCompile(` ([^.]+)`).FindStringSubmatch(via)
```

Comment:
`([^.]+)` matches lots of characters that are not valid in a subdomain like `,` and `\`. A regex for a subdomain should look like this:

https://github.com/apache/trafficcontrol/blob/dad57721e2a82ad0fb5eb1d30ba2ca33542d68ec/vendor/github.com/go-ozzo/ozzo-validation/is/rules.go#L130

Reply:
had to modify, but learned non capture groups along the way, pretty cool